Package.swift

```diff
@@ -165,36 +172,71 @@ let package = Package(
             ]),
         .target(
             name: "ObjectStore",
-            dependencies: ["Storage", "QueryParser", "SyncClient"],
+            dependencies: ["SyncClient", "QueryParser"],
             path: "src",
             exclude: [
                 "realm/object-store/impl/epoll",
                 "realm/object-store/impl/generic",
                 "realm/object-store/impl/windows",
-                "realm/object-store/c_api",
+                "realm/object-store/c_api"
             ],
             sources: ["realm/object-store"],
             publicHeadersPath: "realm/object-store",
             cxxSettings: ([
                 .define("REALM_ENABLE_SYNC", to: "1"),
                 .define("REALM_PLATFORM_APPLE", to: "1", .when(platforms: [.macOS, .iOS, .tvOS, .watchOS])),
-                .headerSearchPath("realm/object-store")
+                .headerSearchPath("realm/object-store"),
+                .headerSearchPath("external/pegtl/include/tao")
```

Comment:
This seems incorrect? Object store should not have anything which depends on pegtl headers.

Reply:
Might have left this in after experimenting. Will remove.

Reply:
The pegtl headers are needed for the Capi target.

Reply:
The Capi target should not require pegtl either. parser.cpp is the only file which depends on pegtl and that file should only be in the QueryParser target.

Reply:
Apparently `c_api/error.cpp` also depends on it:
<img width="540" alt="Screenshot 2021-01-06 at 20 54 12" src="https://user-images.githubusercontent.com/4356928/103818927-5602b580-5061-11eb-8bee-75bb9504b558.png">

Reply:
Unfortunately including pegtl headers is the only way to detect that an exception was a query parsing failure. :-(

The query parser is undergoing massive changes at the moment though, so I wouldn't worry about it for now.